dev/io.openliberty.security.jakartasec.3.0.internal.cdi/src/io/openliberty/security/jakartasec/cdi/beans/OidcHttpAuthenticationMechanism.java

```diff
@@ -138,6 +163,93 @@ private AuthenticationStatus processStartFlowResult(ProviderAuthenticationResult
+            status = httpMessageContext.notifyContainerAboutLogin(CredentialValidationResult.INVALID_RESULT);
+        } catch (AuthenticationResponseException e) {
+            // Something else beyond spec. defined errors
+        }
+
+        return status;
+    }
+
+    private AuthenticationStatus processContinueFlowResult(ProviderAuthenticationResult providerAuthenticationResult,
+                                                           HttpMessageContext httpMessageContext) throws AuthenticationException {
+        AuthenticationStatus status = AuthenticationStatus.SEND_FAILURE;
+
+        if (providerAuthenticationResult != null) {
+            AuthResult authResult = providerAuthenticationResult.getStatus();
+
+            if (AuthResult.SUCCESS.equals(authResult)) {
+                status = handleOidcLogin(providerAuthenticationResult, httpMessageContext);
+            }
+        }
+
+        return status;
+    }
+
+    private AuthenticationStatus handleOidcLogin(ProviderAuthenticationResult providerAuthenticationResult, HttpMessageContext httpMessageContext) throws AuthenticationException {
+        Credential credential = createOidcTokensCredential(providerAuthenticationResult);
+        // TODO: "When available in the "Token Response", the optional fields "refresh_token" and "expires_in" must be stored internally." Store full credential in the subject.
+        return validateCredentials(credential, httpMessageContext);
+    }
+
+    private Credential createOidcTokensCredential(ProviderAuthenticationResult providerAuthenticationResult) {
+        Credential credential = null;
+
+        Hashtable<String, Object> customProperties = providerAuthenticationResult.getCustomProperties();
+        if (customProperties != null) {
+            TokenResponse tokenResponse = (TokenResponse) customProperties.get("TOKEN_RESPONSE");
+            if (tokenResponse != null) {
+                // TODO: credential = new OidcTokensCredential(client, tokenResponse, userinfoResponse);
+                credential = new Credential() {
+                    {
+                    }
+                };
+            }
+        }
+
+        return credential;
+    }
+
+    private AuthenticationStatus validateCredentials(Credential credential, HttpMessageContext httpMessageContext) throws AuthenticationException {
+        int rspStatus;
+        String issuer = JavaEESecConstants.DEFAULT_REALM; // TODO: Set to "iss" claim from the identity token.
+        AuthenticationStatus status = utils.handleAuthenticate(getCDI(), issuer, credential, httpMessageContext.getClientSubject(), httpMessageContext);
+
+        if (status == AuthenticationStatus.SUCCESS) {
+            Map<String, Object> messageInfoMap = httpMessageContext.getMessageInfo().getMap();
+            messageInfoMap.put("jakarta.servlet.http.authType", "JAKARTA_OIDC");
+            messageInfoMap.put("jakarta.servlet.http.registerSession", Boolean.TRUE.toString());
```

Comment:
we also set this in utils.handleAuthenticate() - probably , need to remove this in oidc flow?
httpMessageContext.getMessageInfo().getMap().put("javax.servlet.http.authType", "JASPI_AUTH");

Reply:
Hi Aruna,
Thank you. There's a getAuthType() in the HttpServletRequest that apps can use to get the auth type. The auth type is overridden to distinguish from other types.